src/webgpu/api/validation/encoding/beginComputePass.spec.ts

```diff
@@ -0,0 +1,57 @@
+export const description = `
+Tests for validation in beginComputePass and GPUComputePassDescriptor as its optional descriptor.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ValidationTest } from '../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('compute_pass_descriptor_location')
+  .desc(
+    `
+  Test that entries in timestampWrites do not have the same location in GPUComputePassDescriptor.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('locationA', ['beginning', 'end'] as const)
+      .combine('locationB', ['beginning', 'end'] as const)
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase(['timestamp-query']);
+  })
+  .fn(async t => {
+    const { locationA, locationB } = t.params;
+
+    const querySet = t.device.createQuerySet({
+      type: 'timestamp',
+      count: 1,
```

Comment:
Hmm... this seems wrong, how can we use queryIndex 1 if there's only one entry in the query set? Shouldn't this cause the test to fail because `beginComputePass( 0 beginning, 1 end )` is expected to be valid but won't be? Are we missing validation in `beginComputePass`?

Looks like the render pass version has the same problem - sorry I totally missed that before. You can fix both in this PR, I think.

Reply:
Ah, it's likely I missed catching this problem because the test has not generated a validation error when the locations are the same. I thought that Chromium doesn't generate a validation error when the locations are the same. Ok, I update the problem for both beginComputePass and render_pass_descriptor.

Reply:
Does Chromium pass the tests now or is it still missing validation for duplicate locations?

Anyway, yeah, it's tricky to write tests when the implementation isn't correct either. It's good to look closely at the error messages and which cases fail to make sure they're the ones you expect given any known implementation bugs. I think that would have caught this because you should have seen an error about writing to a query set index that's out of bounds. At least, hopefully.